### PR TITLE
NSX-T DHCP Binding support

### PR DESCRIPTION
**Pull request by Didainius**
This PR adds support for NSX-T DHCP Binding configuration in VCD 10.3.1+ versions. It will be used in PR https://github.com/vmware/terraform-provider-vcd/pull/1039

Tested in:
10.3.0, 10.3.1, 10.4.1

### CODE REVIEW

File: govcd/openapi_org_network_dhcp_binding.go [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointOrgVdcNetworksDhcpBindings
+	apiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(fmt.Sprintf(endpoint, orgVdcNet.OpenApiOrgVdcNetwork.ID))
+	if err != nil {
+		return nil, err
+	}
```
Comment (lvirbalas) [inline govcd/openapi_org_network_dhcp_binding.go:37]: Just a thought for the future: maybe we should think how to encapsulate this block into a single call, as it is repeating in different places. On the other hand, being explicit has some benefits too.

**Review by lvirbalas — APPROVED** [review]

**Review by adezxc — APPROVED** [review]

File: govcd/openapi_org_network_dhcp_binding.go [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+	typeResponses := []*types.OpenApiOrgVdcNetworkDhcpBinding{{}}
+	err = client.OpenApiGetAllItems(apiVersion, urlRef, queryParameters, &typeResponses, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Wrap all typeResponses into OpenApiOrgVdcNetwork types with client
```
Comment (adambarreiro) [inline govcd/openapi_org_network_dhcp_binding.go:95]:
```suggestion
	// Wrap all typeResponses into OpenApiOrgVdcNetworkDhcpBinding types with client
```

**Review by adambarreiro — APPROVED** [review]

File: govcd/openapi_org_network_dhcp_binding.go (hunk 1 above)
Comment (Didainius) [reply govcd/openapi_org_network_dhcp_binding.go:37]: I have some thoughts to experiment with generics for reusing a lot of it, but needs some experimentation. In some cases, we have to do some "custom" things (mainly handling some "one off" VCD API behaviors).

File: govcd/openapi_org_network_dhcp_binding.go (hunk 2 above)
Comment (Didainius) [reply govcd/openapi_org_network_dhcp_binding.go:95]: Fixed.

**Pull request merged**